### ISSUE TITLE
Cannot change password via ESC menu after using /setpassword command

### ISSUE THREAD

**Issue by sorcerykid** [minetest/minetest#10331]
##### Minetest version
```
Minetest 5.3-8ebeed5
```

##### OS / Hardware

Operating system: Centos 6.8 x86_64
CPU: Intel(R) Xeon(R) CPU E3-1231 v3 @ 3.40GHz

##### Summary
When resetting a player's password using the `/setpassword` chat command, the player is no longer able to reset the password from the ESC menu until logging off from the server. The following error messages are displayed in debug.txt

First attempt:
```
2020-08-24 20:07:11: ACTION[Server]: Server: User sorcerykid at XX.XX.XX.XX tried to change their password, but supplied wrong (SRP) password for authentication.
```

Second attempt:
```
2020-08-24 20:08:16: ACTION[Server]: Server: got SRP _A packet, while auth is already going on with mech 2 from XX.XX.XX.XX (wantSudo=1). Ignoring.
```

I became aware of this bug while testing my own authentication handler. However, reverting to the builtin authentication handler yields the same results.

##### Steps to reproduce
1) Login to the sever as administrator
1) Issue command `/setpassword <admin_username> happy`
2) Hit ESC and choose 'Change Password'
3) Fill in the blanks accordingly, then press 'Change'

**Comment by SmallJoker** [comment]
If the password can only be changed after logging in with the last set one, then it might have the same origin as bug #5954.

**Comment by sorcerykid** [comment]
Thanks for the heads up regarding that issue. It's hard for me to say whether that's related since I'm not familiar with SRP. All I can say for certain is that my situation definitely occurs after using the `/setpassword` command.

**Comment by Zughy** [comment]
Tested in 5.5, I can confirm, but this is not related to the `/setpassword` command. The Change password formspec works only the first time you change the password. So it doesn't matter if the first time you use `/setpassword`, if an admin uses it on you, or if you use the formspec. The second time won't anyway

**Comment by sfan5** [comment]
fixed by 00f71c3b9d35e1cdd5aa62491a46068358aa8b2a
